### PR TITLE
Add the `cut-body-after` configuration option

### LINKED ISSUE #200: Crop off point for pull requests
Add a bors.toml directive to the effect of:

```toml
cut-description-after = "<!-- Reviewable:start -->"
```

If present, then when bors writes PR bodies into its merge commit, it will ignore the text blob and everything after it.

### PR DESCRIPTION
**Pull request by notriddle**
Fixes #200

### CODE REVIEW

**Comment by notriddle** [comment]
This should be in a reviewable state now.

File: apps/bors_github/lib/github.ex [hunk 1]
```diff
@@ -62,7 +62,7 @@ defmodule BorsNG.GitHub do
     from: bitstring,
     to: bitstring,
     commit_message: bitstring,
-    }) :: %{commit: bitstring, tree: binary} | :conflict
+    }) :: %{commit: binary, tree: binary} | :conflict
```
Comment (Macarse) [inline apps/bors_github/lib/github.ex]: Why was this change needed?

File: apps/bors_worker/test/batcher/message_test.exs [hunk 2]
```diff
@@ -50,16 +50,51 @@ defmodule BorsNG.Worker.BatcherMessageTest do
...
+    - [ ] Run the linter
+    - [ ] Run any new or changed tests
+    - [ ] This PR fixes #___ (fill in if it exists)
+    - [ ] Make sure your commit messages make sense
+    """
+    patches = [
+      %{patch: %{pr_xref: 1, title: title, body: body}, reviewer: "bill"} ]
+    actual_message = Message.generate_commit_message(
+      patches,
+      "\n\n<!-- boilerplate follows -->")
```
Comment (Macarse) [inline apps/bors_worker/test/batcher/message_test.exs]: Would bors allow to use `\n` inside the cut off token?

File: apps/bors_worker/test/batcher/message_test.exs [hunk 3]
```diff
@@ -50,16 +50,51 @@ defmodule BorsNG.Worker.BatcherMessageTest do
...
+    assert expected_message == actual_message
+  end
 end
```
Comment (Macarse) [inline apps/bors_worker/test/batcher/message_test.exs]: For completeness, I would add a test for more than one match and another with no matches at all.

File: apps/bors_github/lib/github.ex (hunk 1 above)
Comment (notriddle) [inline apps/bors_github/lib/github.ex]: It actually wasn't needed. The reason I did it is because [`binary` is a more specific typespec than `bitstring`](https://hexdocs.pm/elixir/typespecs.html#types-and-their-syntax). Namely, if you assert that the type is `binary`, you assert that the number if bits in the bitstring is a multiple of 8 (which is always true for UTF-8).

File: apps/bors_worker/test/batcher/message_test.exs (hunk 2 above)
Comment (notriddle) [inline apps/bors_worker/test/batcher/message_test.exs]: Yeah, it would.

File: apps/bors_worker/test/batcher/message_test.exs (hunk 3 above)
Comment (notriddle) [inline apps/bors_worker/test/batcher/message_test.exs]: Done.

**Comment by notriddle** [comment]
BTW, you probably don't want to merge it until I've squashed the `fixup!` commit into the main commit.

File: apps/bors_worker/lib/batcher/message.ex [hunk 4]
```diff
@@ -71,20 +71,30 @@ defmodule BorsNG.Worker.Batcher.Message do
...
+      #{body}
       """
     end)
     "#{commit_title}\n#{commit_body}"
   end
 
+  def cut_body(body, nil), do: body
+  def cut_body(body, cut) do
+    body
+    |> String.splitter(cut)
```
Comment (Macarse) [inline apps/bors_worker/lib/batcher/message.ex]: while searching about `String.splitter` I found this: https://github.com/elixir-lang/elixir/issues/6148

**Comment by notriddle** [comment]
bors r= @Macarse

**Comment by notriddle** [comment]
As discussed in IRC, switched to `Enum.at(0)` instead of a combination of `Enum.take(1) |> hd()`

File: apps/bors_worker/lib/batcher/message.ex [hunk 5]
```diff
@@ -71,20 +71,30 @@ defmodule BorsNG.Worker.Batcher.Message do
...
+    |> Enum.take(1)
+    |> hd()
```
Comment (Macarse) [inline apps/bors_worker/lib/batcher/message.ex]: Talked to @notriddle privately. this can be done without the `hd()`. Using `Enum.at(0)`.
